instrumentation/elasticsearch/elasticsearch-rest-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/elasticsearch/rest/ElasticsearchClientAttributeExtractor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.elasticsearch.rest;
+
+import static io.opentelemetry.instrumentation.api.internal.AttributesExtractorUtil.internalSet;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.network.internal.NetworkAttributes;
+import io.opentelemetry.instrumentation.api.instrumenter.url.internal.UrlAttributes;
+import io.opentelemetry.instrumentation.api.internal.SemconvStability;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import javax.annotation.Nullable;
+import org.apache.http.HttpHost;
+import org.elasticsearch.client.Response;
+
+public class ElasticsearchClientAttributeExtractor
+    implements AttributesExtractor<ElasticsearchRestRequest, Response> {
+
+  private static final String PATH_PARTS_ATTRIBUTE_PREFIX = "db.elasticsearch.path_parts.";
+
+  private static void setServerAttributes(AttributesBuilder attributes, Response response) {
+    HttpHost host = response.getHost();
+    if (host != null) {
+      if (SemconvStability.emitStableHttpSemconv()) {
+        internalSet(attributes, NetworkAttributes.SERVER_ADDRESS, host.getHostName());
+        internalSet(attributes, NetworkAttributes.SERVER_PORT, (long) host.getPort());
+      }
+      if (SemconvStability.emitOldHttpSemconv()) {
+        internalSet(attributes, SemanticAttributes.NET_PEER_NAME, host.getHostName());
+        internalSet(
+            attributes, SemanticAttributes.NET_PEER_PORT, (long) host.getPort());
+      }
+    }
+  }
+
+  private static void setUrlAttribute(AttributesBuilder attributes, Response response) {
+    String uri = response.getRequestLine().getUri();
+    uri = uri.startsWith("/") ? uri : "/" + uri;
+    String fullUrl = response.getHost().toURI() + uri;
+
+    if (SemconvStability.emitStableHttpSemconv()) {
+      internalSet(attributes, UrlAttributes.URL_FULL, fullUrl);
+    }
+
+    if (SemconvStability.emitOldHttpSemconv()) {
+      internalSet(attributes, SemanticAttributes.HTTP_URL, fullUrl);
+    }
+  }
+
+  private static void setPathPartsAttributes(
+      AttributesBuilder attributes, ElasticsearchRestRequest request) {
+    ElasticsearchEndpointDefinition endpointDef = request.getEndpointDefinition();
+    if (endpointDef == null) {
+      return;
+    }
+
+    endpointDef.processPathParts(
+        request.getEndpoint(),
+        (key, value) -> {
+          String attributeKey = PATH_PARTS_ATTRIBUTE_PREFIX + key;
+          internalSet(attributes, AttributeKey.stringKey(attributeKey), value);
+        });
+  }
+
+  @Override
+  public void onStart(
```

Comment:
Nit: would you mind putting the public methods on top, so that you can read the class top-down?

Reply:
Oof, didn't expect that 🤦‍♂️
It's fine to leave them as they are then, let's not fight the formatter 😅